src/index.tsx

```diff
@@ -97,13 +121,14 @@ function LazyHydrate(props: Props) {
     }
 
     if (whenIdle) {
-      // @ts-ignore
-      if (requestIdleCallback) {
-        // @ts-ignore
-        const idleCallbackId = requestIdleCallback(hydrate, { timeout: 500 });
+      if (window.requestIdleCallback) {
+        const idleCallbackId = window.requestIdleCallback(hydrate, {
```

Comment:
don't use window, it can cause problems with server builds. use requestIdleCallback directly